app/actions/nodeStorageActions.js

```diff
@@ -1,17 +1,78 @@
 // @flow
 import { createActions } from 'spunky'
 import { isEmpty } from 'lodash-es'
+import { rpc } from 'neon-js'
 
 import { getStorage, setStorage } from '../core/storage'
+import {
+  MAIN_NETWORK_ID,
+  TEST_NETWORK_ID,
+  NODES_MAIN_NET,
+  NODES_TEST_NET,
+  NODE_EXLUSION_CRITERIA
+} from '../core/constants'
+import { findNetworkByLabel } from '../core/networks'
 
 const ID = 'nodeStorage'
 const STORAGE_KEY = 'selectedNode'
+let cachedRPCUrl
 
 type Props = {
   url: string,
   net: string
 }
 
+export const getRPCEndpoint = async (
+  net: string,
+  excludeCritera: Array<string> = NODE_EXLUSION_CRITERIA
+) => {
+  try {
+    if (cachedRPCUrl) return cachedRPCUrl
+    const NETWORK = findNetworkByLabel(net)
+    let nodeList
+    switch (NETWORK.id) {
```

Comment:
are we sure that `NETWORK` will contain an `id` ?

Reply:
It must contain an ID because no matter what a NetworkItemType is always returned 
```
export const findNetworkByLabel = (networkLabel: string): NetworkItemType => {
  const networks = getNetworks()
  return networks.find(({ label }) => networkLabel === label) || networks[0]
}
```